src/index.tsx

```diff
@@ -154,87 +155,94 @@ export default class ReactToPrint extends React.Component<IReactToPrintProps> {
+                                for (let j = 0, cssLen = sheet.cssRules.length; j < cssLen; ++j) {
+                                    if (typeof sheet.cssRules[j].cssText === "string") {
+                                        styleCSS += `${sheet.cssRules[j].cssText}\r\n`;
+                                    }
                                 }
+                                newHeadEl.setAttribute("id", `react-to-print-${i}`);
+                                newHeadEl.appendChild(domDoc.createTextNode(styleCSS));
+                                domDoc.head.appendChild(newHeadEl);
                             }
-                            newHeadEl.setAttribute("id", `react-to-print-${i}`);
-                            newHeadEl.appendChild(domDoc.createTextNode(styleCSS));
-                            domDoc.head.appendChild(newHeadEl);
-                        }
-                    } else {
-                        // Many browsers will do all sorts of weird things if they encounter an
-                        // empty `href` tag (which is invalid HTML). Some will attempt to load the
-                        // current page. Some will attempt to load the page"s parent directory.
-                        // These problems can cause `react-to-print` to stop  without any error
-                        // being thrown. To avoid such problems we simply do not attempt to load
-                        // these links.
-                        if (node.hasAttribute("href") && !!node.getAttribute("href")) {
-                            const newHeadEl = domDoc.createElement(node.tagName);
+                        } else {
+                            // Many browsers will do all sorts of weird things if they encounter an
+                            // empty `href` tag (which is invalid HTML). Some will attempt to load the
+                            // current page. Some will attempt to load the page"s parent directory.
+                            // These problems can cause `react-to-print` to stop  without any error
+                            // being thrown. To avoid such problems we simply do not attempt to load
+                            // these links.
+                            if (node.hasAttribute("href") && !!node.getAttribute("href") ||
+                                (node.hasAttribute('src') && !!node.getAttribute('src'))
+                            ) {
+                                const newHeadEl = domDoc.createElement(node.tagName);
+
+                                // node.attributes has NamedNodeMap type that is not an Array and can be
+                                // iterated only via direct [i] access
+                                for (let j = 0, attrLen = node.attributes.length; j < attrLen; ++j) {
+                                    const attr = node.attributes[j];
+                                    if (attr && attr.nodeValue) {
```

Comment:
This check needs to be more specific. The empty string `""` is a valid input for the value attribute: https://developer.mozilla.org/en-US/docs/Web/API/Element/setAttribute

Reply:
Please help to check